### PR TITLE
Add domain to the deploy config.

### PR DESCRIPTION
**Pull request by cseed**
Modify hailctl dev config to let you set the domain.  Note, this is an
interface change since I changed `hailctl dev config` to act like
gcloud/kubectl `... set property value`.

The hardest part of this was getting a doubly-nested subcommand to work in argprase.  The existing code is wack, but I will change everything to work like hailctl dev/dev config does below.

### CODE REVIEW

File: hail/python/hailtop/hailctl/dev/config/set_property.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                        help="Value to set property to.")
+
+
+def main(args):
+    deploy_config = get_deploy_config()
+    config = deploy_config.get_config()
+
+    p = args.property
+    if p == 'default':
+        p = 'default_namespace'
```
Comment (danking) [inline hail/python/hailtop/hailctl/dev/config/set_property.py:21]: I don't like this, it's a confusing name and this is extra complexity. I'm pretty sure all of us have canned scripts / history lines for `hailctl dev config` anyway, let's go with `hailctl dev config set default_namespace ...`.

**Review by danking — CHANGES_REQUESTED** [review]
One comment and this:

While verifying this I found a bad configuration in web_common:
```
../../../../web_common/web_common/templates/header.html:37:        <a class="header-dropdown-menu-link" href="https://hail.is/docs/batch/">Docs</a>
```
Can you fix that while you're in here?

File: hail/python/hailtop/hailctl/dev/config/set_property.py (hunk 1 above)
Comment (cseed) [reply hail/python/hailtop/hailctl/dev/config/set_property.py:21]: Agreed, fixed.

**Comment by cseed** [comment]
The web_common fix isn't totally trivial.  I will PR that separately.

**Review by danking — APPROVED** [review]

**Pull request merged**